src/main/java/uk/gov/hmcts/reform/sscs/ccd/service/CcdService.java

```diff
@@ -87,16 +87,15 @@ public SscsCaseDetails findCcdCaseByNinoAndBenefitTypeAndMrnDate(SscsCaseData ca
     public SscsCaseDetails updateSubscription(String appealNumber, String email, IdamTokens idamTokens) {
         try {
             SscsCaseDetails caseDetails = getCaseByAppealNumber(appealNumber, idamTokens);
-
             if (caseDetails != null) {
                 SscsCaseData caseData = caseDetails.getData();
-
                 String subscribeEmail = null != email ? YES : NO;
-
-                if (appealNumber.equals(caseData.getSubscriptions().getAppellantSubscription().getTya())) {
+                if (caseData.getSubscriptions().getAppellantSubscription() != null && appealNumber.equals(caseData.getSubscriptions().getAppellantSubscription().getTya())) {
                     updateAppellantSubscription(email, caseData, subscribeEmail);
-                } else if (appealNumber.equals(caseData.getSubscriptions().getRepresentativeSubscription().getTya())) {
+                } else if (caseData.getSubscriptions().getRepresentativeSubscription() != null && appealNumber.equals(caseData.getSubscriptions().getRepresentativeSubscription().getTya())) {
                     updateRepresentativeSubscription(email, caseData, subscribeEmail);
+                } else if (caseData.getSubscriptions().getAppointeeSubscription() != null && appealNumber.equals(caseData.getSubscriptions().getAppointeeSubscription().getTya())) {
```

Comment:
Let's take appointee out from here because it's out of the ticket scope. 
And le'ts check with Ify to see if we have to create a ticket to be able to update appointee subscriptions at all.